### ISSUE TITLE
build issues with aclocal

### ISSUE THREAD

**Issue by forslund** [MycroftAI/mimic#68]
Getting home from vacation trying to build the mimic development branch I got the following error
```
CDPATH="${ZSH_VERSION+.}:" && cd .. && /bin/bash /home/ake/projects/c/mimic/config/missing aclocal-1.14 -I m4
/home/ake/projects/c/mimic/config/missing: line 81: aclocal-1.14: command not found
WARNING: 'aclocal-1.14' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <http://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/autoconf>
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
Makefile:1752: recipe for target '../aclocal.m4' failed
make: *** [../aclocal.m4] Error 127
```
This is because aclocal is version 1.15 on my ubuntu system and not 1.14. On my debian VM I get the same issue with a fresh checkout of the project unless I install automake.

The Makefile seem to include rules for building the `aclocal.m4` and the dependencies may be considered updated when checked out and this trigger the error in question.

I see two solutions:
1: remove the files generated by `autogen.sh`  and the procedure for building shall include running `./autogen.sh` (I've prepared a branch for this named *autogen*)

2: The rules should exclude `aclocal.m4` (This I haven't quite figured out how to accomplish)

**Comment by zeehio** [comment]
I vote for solution 1.

Regarding solution 2, I think it can be implemented using `AM_MAINTAINER_MODE([disable])` in `configure.ac` but that approach is no longer recommended

**Comment by forslund** [comment]
That's my preferred solution as well.

I'll look over the branch I created for this and create a PR tonight.

**Comment by zeehio** [comment]
If we are going towards removing the autotools generated files then all config/*, configure, Makefile.in, aclocal.m4, m4/* can be removed as well as they can be generated from the autogen script with

**Comment by forslund** [comment]
Sure thing. Thanks for the list I was under the impression that the m4-dir was a prerequisite and would have missed it.
*...Updating and cleaning up...*

**Comment by zeehio** [comment]
I believe it is also possible to remove `include/config.h.in` if we add `autoheader` in the autogen.sh script. I am not sure, but I think adding autoheader right after aclocal should work fine.

**Comment by forslund** [comment]
Yes that seems to work fine. Now I just need to fix the OS X travis build.

**Comment by forslund** [comment]
There, it builds but maybe `config.guess` should be kept? I'm getting a warning in travis that it's missing

**Comment by rhdunn** [comment]
The `config.guess` and `config.sub` files comes from `automake`. You could do something like:

    if [ x${AUTOMAKE_DIR} = x ] ; then
    	AUTOMAKE_DIR=`find /usr/share/automake-* -maxdepth 0 -type d 2>&1 | sort | tail -n 1`
    fi

    if [ ! -e "${AUTOMAKE_DIR}" -o ! -e "${AUTOMAKE_DIR}/install-sh" ] ; then
    	echo "Unable to locate the automake directory."
    	echo "Set AUTOMAKE_DIR to the correct automake directory path."
   	exit 1
    fi

    ln -sf ${AUTOMAKE_DIR}/config.guess .
    ln -sf ${AUTOMAKE_DIR}/config.sub .

**Comment by forslund** [comment]
Hmm, after reading and testing some more I think `config.guess` and `config.sub` should be taken care of by `automake --add-missing` (Locally it seems to work)

**Comment by forslund** [comment]
Ok, now I get it. The file is there after `./autogen.sh` is run but the `run_test.sh` script runs config.guess directly at startup, before `./autogen.sh` had been run. Everything is fine, only some slight juggling of the execution order is required.

**Comment by zeehio** [comment]
Closed in #69
